script/setup.js

```diff
@@ -113,36 +121,75 @@ try {
 
 	const skipApi = values["skip-api"];
 
-	console.log();
-	console.log(chalk.gray`Hydrating package metadata locally...`);
+	const successSpinnerBlock = (blockText) => {
+		s.start(chalk.green("✅ " + blockText));
+		s.stop(chalk.green("✅ " + blockText));
+	};
+
+	const skipSpinnerBlock = (blockText) => {
+		s.start(chalk.yellow("➖ " + blockText));
+		s.stop(chalk.yellow("➖ " + blockText));
+	};
+
+	successSpinnerBlock("Started hydrating package metadata locally.");
 
 	async function readFileAsJSON(filePath) {
-		return JSON.parse((await fs.readFile(filePath)).toString());
+		try {
+			return JSON.parse((await readFile(filePath)).toString());
+		} catch (error) {
+			throw new Error(
+				`Could not read file from ${filePath} as JSON. Please ensure the file exists and is valid JSON.`,
+				{ cause: error }
+			);
+		}
 	}
 
-	const [existingContributors, existingPackage, outcomeLabels] =
-		await Promise.all([
-			readFileAsJSON("./.all-contributorsrc"),
-			readFileAsJSON("./package.json"),
-			readFileAsJSON("./script/labels.json"),
-		]);
-
-	await fs.writeFile(
-		"./.all-contributorsrc",
-		prettier.format(
-			JSON.stringify({
-				...existingContributors,
-				contributors: [
-					{
-						...existingContributors.contributors.find(
-							({ login }) => login === "JoshuaKGoldberg"
-						),
-						contributions: ["tool"],
-					},
-				],
-			}),
-			{ parser: "json" }
-		)
+	const pSpinner = async (callback, options) => {
```

Comment:
[Naming] I'm not a huge fan of abbreviations. It can be hard to map their names back to what they stand for. How about `withSpinner`?

```suggestion
	const startSpinner = async (callback, options) => {
```

Reply:
Same with me, React mind asked me to name `useSpinner` but I preferred to use `sindresorhus` naming of promise variable naming.